### PR TITLE
Resourcify Badge Achievements

### PR DESCRIPTION
**Pull request by Zhao-Andy**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
`resourcify` badge achievements so we can allow `user.add_role :single_resource_admin, BadgeAchievement` to be a thing. Needed for internal operations mostly.

Also cleans up the constants' syntax a bit for easier writing

## Related Tickets & Documents
https://github.com/orgs/forem/projects/27#card-55779138

## QA Instructions, Screenshots, Recordings
1. Visit `/admin/users/1/edit`
2. Add the Badge and BadgeAchievement roles
3. See that they've been added properly

### UI accessibility concerns?
Nope

## Added tests?

- [x] Yes

## [Forem core team only] How will this change be communicated?

- [x] I will share this change internally with the appropriate teams

## [optional] Are there any post deployment tasks we need to perform?
No

## [optional] What gif best describes this PR or how it makes you feel?

The letter B

### CODE REVIEW

File: app/lib/constants/role.rb [hunk 1]
```diff
@@ -1,27 +1,33 @@
...
+    BASE_ROLES = [
+      "Warn",
+      "Comment Suspend",
+      "Suspend",
+      "Regular Member",
+      "Trusted",
+      "Pro",
+    ].freeze
 
-    SPECIAL_ROLES = ["Admin",
```
Comment (Zhao-Andy) [inline app/lib/constants/role.rb:10]: Just moved the first and last brackets in their own line, and somewhat alphabetized the list.

**Review by rhymes — APPROVED** [review]

File: app/lib/constants/role.rb [hunk 2]
```diff
@@ -1,27 +1,33 @@
...
-                  "Regular Member",
-                  "Trusted",
-                  "Pro"].freeze
+    BASE_ROLES = [
+      "Warn",
+      "Comment Suspend",
+      "Suspend",
+      "Regular Member",
+      "Trusted",
+      "Pro",
```
Comment (rhymes) [inline app/lib/constants/role.rb:9]: now that I saw it in the diff, I wonder, is `Pro` still a thing?

Comment (Zhao-Andy) [reply app/lib/constants/role.rb:9]: I think we technically still have the functionality behind the `Pro` role 😅

Comment (rhymes) [reply app/lib/constants/role.rb:9]: Ah yeah, the analytics! I thought we renamed that to `analytics` for some reason

**Review by jacobherrington — APPROVED** [review]

**Pull request merged**